### PR TITLE
Added rules and telemeter query for storage provisioners usage metrics

### PR DESCRIPTION
**Pull request by anmolsachan**
1) Adds two rules to the cluster monitoring operator promethues rules, which will give the total amount of storage requested or used by PVCs from a particular storage provisioner
2) Adds query for the above rules to be sent via telemeter.

Why: This will help in 
- planning how to better support storage on OCP, where OCS can grow, where OCS is being used heavily in cloud environments.
- aggregating statistics on different storage providers  on OCP and their usage.

Metric results look like: 
cluster:kube_persistentvolumeclaim_resource_requests_storage_bytes:provisioner:sum
Screenshot from 2020-07-30 20-03-32

cluster:kubelet_volume_stats_used_bytes:provisioner:sum
Screenshot from 2020-07-30 20-02-51

Signed-off-by: Anmol Sachan <anmol13694@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: assets/prometheus-k8s/rules.yaml [hunk 1]
```diff
@@ -866,6 +866,14 @@ spec:
...
         label_node_role_kubernetes_io)
       record: cluster:node_instance_type_count:sum
+    - expr: sum  by (provisioner) (kube_persistentvolumeclaim_resource_requests_storage_bytes
+        * on (namespace,persistentvolumeclaim) group_right() (kube_persistentvolumeclaim_info
+        * on (storageclass)  group_left(provisioner) kube_storageclass_info))
+      record: cluster:storage_provisioner_capacity_requested:total
+    - expr: sum  by (provisioner) (kubelet_volume_stats_used_bytes * on (namespace,persistentvolumeclaim)
+        group_right() (kube_persistentvolumeclaim_info * on (storageclass)  group_left(provisioner)
+        kube_storageclass_info))
+      record: cluster:storage_provisioner_capacity_used:total
```
Comment (lilic) [inline assets/prometheus-k8s/rules.yaml:876]: Recording rules have certain naming rules see docs for this https://prometheus.io/docs/practices/rules/

`cluster:kubelet_volume_stats_used_bytes:provisioner:sum`

 would be more appropriate here

Comment (lilic) [reply assets/prometheus-k8s/rules.yaml:876]: Same goes with the other recording rule.

File: Documentation/data-collection.md [hunk 2]
```diff
@@ -145,7 +145,14 @@ data:
     # version, phase, reason labels. When a csv is updated, the previous time
     # series associated with the csv will be deleted.
     - '{__name__="csv_abnormal"}'
-    # OCS metrics to be collected:
+    # Storage metrics:
+    # cluster:storage_provisioner_capacity_requested:total will give the total amount of
```
Comment (s-urbaniak) [inline Documentation/data-collection.md:149]: do you mind to add team owners for the new team metrics? there seems to be just one entry prefexing the ceph ones.

Comment (anmolsachan) [reply Documentation/data-collection.md:149]: Sure.

File: assets/prometheus-k8s/rules.yaml (hunk 1 above)
Comment (anmolsachan) [reply assets/prometheus-k8s/rules.yaml:876]: ack, I will update the PR, Thanks

Comment (anmolsachan) [reply assets/prometheus-k8s/rules.yaml]: @lilic I have made the changes.

File: Documentation/data-collection.md [hunk 3]
```diff
@@ -145,7 +145,14 @@ data:
     # version, phase, reason labels. When a csv is updated, the previous time
     # series associated with the csv will be deleted.
     - '{__name__="csv_abnormal"}'
-    # OCS metrics to be collected:
+    # (OCS-storage-management-team) Generic Storage metrics:
```
Comment (s-urbaniak) [inline Documentation/data-collection.md:148]: oh please add a github alias.

Comment (anmolsachan) [reply Documentation/data-collection.md]: @s-urbaniak There is no particular alias for our team, we contribute to multiple ones. I have added @openshift/ocs-operator, it is part of OCP.

**Comment by anmolsachan** [comment]
@s-urbaniak @lilic All the review comments have been addressed, please review. Thanks.

**Comment by lilic** [comment]
/lgtm
/hold

cc @smarterclayton for 👍

**Comment by s-urbaniak** [comment]
@anmolsachan needs rebase, else lgtm from me modulo final verification by @smarterclayton

**Comment by anmolsachan** [comment]
@s-urbaniak I have updated the PR. 
Can you please the LGTM label. It got removed due to the rebase. Thanks

@smarterclayton Please review.Thanks

**Comment by smarterclayton** [comment]
Approved, thank you for broadening this

**Comment by lilic** [comment]
/hold cancel

You might need a bugzilla for this, not sure if its enforced already lets see. :D

**Pull request merged**